posthog/src/main/java/com/posthog/java/PostHog.java

```diff
@@ -194,4 +199,41 @@ private JSONObject getEventJson(String event, String distinctId, Map<String, Obj
         }
         return eventJson;
     }
+
+    /**
+     * 
+     * @param featureFlag which uniquely identifies your feature flag
+     *
+     * @param distinctId which uniquely identifies your user in your database. Must
+     *                   not be null or empty.
+     */
+    public boolean isFeatureFlagEnabled(String featureFlag, String distinctId) {
+        JSONObject response = sender.post("/decide/?v=3", distinctId);
+
+        try {
+            return response.getJSONObject("featureFlags").getBoolean(featureFlag);
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+        return false;
+    }
+
+    /**
+     * 
+     * @param featureFlag which uniquely identifies your feature flag
+     *
+     * @param distinctId which uniquely identifies your user in your database. Must
+     *                   not be null or empty.
+     */
+    public String getFeatureFlag(String featureFlag, String distinctId) {
```

Comment:
Hey @batosti , the interface is a bit wrong here:

1. `getFeatureFlag` returns the variant
2. `getFeatureFlagPayload` returns the payload

So, I'd recommend seeing what posthog-python does: https://github.com/PostHog/posthog-python/blob/master/posthog/client.py#L495 - isFeatureEnabled uses the output of getFeatureFlag, and getFeatureFlag returns the payload

Reply:
Hello @neilkakkar, I check the python version. It's a little bit complex, so I try to implement a simple solution. So I can call the /decide endpoint on getFeatureFlag to return the payload and isFeatureEnabled can use the getFeatureFlag?

Reply:
yes, that's what I meant, don't do the complex handling for local evaluation, just copy the function signature and call `/decide` 👍

Reply:
Hello @neilkakkar, 

I made some changes, and create a local copy, it's easier than use the same call for decide, because some flags have payload and others don't. Check if it's ok and I'll write the unit tests for then.

Thanks and Regards